### PR TITLE
Protect save and update from CSRF

### LINKED ISSUE #116: Cover more actions with XSRF token protection
Delete is covered, but we could cover more.

### PR DESCRIPTION
**Pull request by noncombatant**
Also, fix a lint.

Fixes #116

### CODE REVIEW

File: tmpl/help.html [hunk 1]
```diff
@@ -122,7 +122,7 @@ <h2 id="api">Application Programming Interface (API)</h2>
 <p>
 Create a new link by sending a POST request with a <code>short</code> and <code>long</code> value:
 
-<pre>{{`$ curl -L -d short=cs -d long=https://cs.github.com/ go
+<pre>{{`$ curl -L -H "Sec-Golink: 1" -d short=cs -d long=https://cs.github.com/ go
```
Comment (creachadair) [inline tmpl/help.html:125]: This is fine as-written, but we could avoid the need for quotes by compacting the arg:

```bash
curl -L -H sec-golink:1 -d short=cs -d long=https://cs.github.com/ go
```

or similar. YMMV.

File: golink.go [hunk 2]
```diff
@@ -259,6 +269,14 @@ type homeData struct {
 	Short  string
 	Long   string
 	Clicks []visitData
+	XSRF   string
+}
+
+// deleteData is the data used by the deleteTmpl template.
```
Comment (creachadair) [inline golink.go:275]: Or perhaps more simply "…by deleteTmpl." ?

File: golink.go [hunk 3]
```diff
@@ -39,7 +39,17 @@ import (
...
 
-const defaultHostname = "go"
+const (
+	defaultHostname = "go"
+
+	// Used as a placeholder short name for generating the XSRF defense token,
+	// when creating new links.
+	newShortName = ".new"
+
+	// JavaScript in browsers cannot set this header, per the Fetch Spec
```
Comment (creachadair) [inline golink.go:49]: Maybe also say how we're using it?

```go
// If the caller sends this header set to a non-empty value, we will allow them
// to make the call even without an XSRF token. JavaScript in browser cannot set
// this header, per the [Fetch Spec].
//
// [Fetch Spec]: https://fetch.spec.whatwg.org
```

perhaps.

**Review by creachadair — APPROVED** [review]
A few small suggestions, otherwise LGTM.

File: golink.go (hunk 3 above)
Comment (noncombatant) [reply golink.go:49]: Done!

File: golink.go (hunk 2 above)
Comment (noncombatant) [reply golink.go:275]: Done

File: tmpl/help.html (hunk 1 above)
Comment (noncombatant) [reply tmpl/help.html:125]: Done

**Pull request merged**